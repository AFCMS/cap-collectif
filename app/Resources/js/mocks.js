// @flow
/* eslint-env jest */
import { type FormProps } from 'redux-form';
import { type IntlShape } from 'react-intl';
import type { RelayPaginationProp, RelayRefetchProp } from 'react-relay';

export const $refType: any = null;
export const $fragmentRefs: any = null;

export const intlMock: IntlShape = {
  locale: 'fr-FR',
  formats: {},
  messages: {},
  now: () => 0,
  formatHTMLMessage: message => String(message),
  formatPlural: message => String(message),
  formatNumber: message => String(message),
  formatRelative: message => String(message),
  formatTime: message => String(message),
  formatDate: message => String(message),
  formatMessage: message => String(message.id),
};

export const formMock: FormProps = {
  anyTouched: false,
  array: {
    insert: jest.fn(),
    move: jest.fn(),
    pop: jest.fn(),
    push: jest.fn(),
    remove: jest.fn(),
    removeAll: jest.fn(),
    shift: jest.fn(),
    splice: jest.fn(),
    swap: jest.fn(),
    unshift: jest.fn(),
  },
  asyncValidate: jest.fn(),
  asyncValidating: false,
  autofill: jest.fn(),
  blur: jest.fn(),
  change: jest.fn(),
  clearAsyncError: jest.fn(),
  clearSubmit: jest.fn(),
  destroy: jest.fn(),
  dirty: false,
  dispatch: jest.fn(),
  error: null,
  form: 'formName',
  handleSubmit: jest.fn(),
  initialize: jest.fn(),
  initialized: true,
  initialValues: {},
  invalid: false,
  pristine: true,
  reset: jest.fn(),
  submitting: false,
  submitFailed: false,
  submitSucceeded: false,
  touch: jest.fn(),
  untouch: jest.fn(),
  valid: true,
  warning: null,
};

export const relayPaginationMock: RelayPaginationProp = {
  environment: {
    applyMutation: jest.fn(),
    sendMutation: jest.fn(),
    lookup: jest.fn(),
    sendQuery: jest.fn(),
    subscribe: jest.fn(),
    streamQuery: jest.fn(),
    retain: jest.fn(),
    unstable_internal: {},
  },
  hasMore: () => false,
  isLoading: () => false,
  loadMore: jest.fn(),
  refetchConnection: jest.fn(),
};

export const relayRefetchMock: RelayRefetchProp = {
  environment: {
    applyMutation: jest.fn(),
    sendMutation: jest.fn(),
    lookup: jest.fn(),
    sendQuery: jest.fn(),
    subscribe: jest.fn(),
    streamQuery: jest.fn(),
    retain: jest.fn(),
    unstable_internal: {},
  },
  refetch: jest.fn(),
};
