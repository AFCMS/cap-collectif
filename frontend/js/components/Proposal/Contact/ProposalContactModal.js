// @flow
import React from 'react';
import { Modal } from 'react-bootstrap';
import { Field, reduxForm, SubmissionError } from 'redux-form';
import { connect } from 'react-redux';
import { FormattedMessage } from 'react-intl';
import CloseButton from '~/components/Form/CloseButton';
import SubmitButton from '~/components/Form/SubmitButton';
import component from '~/components/Form/Field';
import { isEmail } from '~/services/Validator';
import type { State, Dispatch } from '~/types';
import FluxDispatcher from '~/dispatchers/AppDispatcher';
import ContactProposalAuthorMutation from '~/mutations/ContactProposalAuthorMutation';

type Props = {|
  ...ReduxFormFormProps,
  show: boolean,
  proposalId: string,
  authorName: string,
  onClose: () => {},
  onSubmit: () => {},
  addCaptchaField: boolean,
|};

type FormValues = {|
  proposalId: string,
  senderName: string,
  message: string,
  replyEmail: string,
  captcha: string,
|};

const onSubmit = (values: FormValues, dispatch: Dispatch, props: Props) => {
  if (!values) return;
  ContactProposalAuthorMutation.commit({ input: { ...values } })
    .then(() => {
      props.reset();
      props.onClose();
      FluxDispatcher.dispatch({
        actionType: 'UPDATE_ALERT',
        alert: {
          bsStyle: 'success',
          content: 'message-sent-with-success',
        },
      });
    })
    .catch(() => {
      throw new SubmissionError({
        _error: 'global.error.server.form',
      });
    });
};

const validate = ({ senderName, message, replyEmail, captcha }: FormValues, props: Props) => {
  const errors = {};

  if (!senderName || senderName?.length < 2) errors.senderName = 'global.required';

  if (!message || message?.length < 2) errors.message = 'global.required';

  if (!replyEmail) {
    errors.replyEmail = 'global.required';
  } else if (!isEmail(replyEmail)) {
    errors.replyEmail = 'global.constraints.email.invalid';
  }

  if (!captcha && props.addCaptchaField && window && window.location.host !== 'capco.test') {
    errors.captcha = 'registration.constraints.captcha.invalid';
  }

  return errors;
};

export const formName = 'ProposalContactModalForm';

export const ProposalContactModal = ({
  handleSubmit,
  show,
  onClose,
  invalid,
  submitting,
  pristine,
  reset,
  authorName,
}: Props) => (
  <Modal
    show={show}
    onHide={() => {
      onClose();
      reset();
    }}
    aria-labelledby="ProposalFormContactModal-modal">
    <form onSubmit={handleSubmit} id={formName}>
      <Modal.Header closeButton>
        <Modal.Title
          id="ProposalFormContactModal-title"
          children={<FormattedMessage id="send-message-to" values={{ messageTo: authorName }} />}
        />
      </Modal.Header>
      <Modal.Body>
        <Field
          label={<FormattedMessage id="your-name" />}
          id="ProposalFormContactModal-senderName"
          name="senderName"
          type="text"
          component={component}
        />
        <Field
          label={<FormattedMessage id="your-email-address" />}
          id="ProposalFormContactModal-replyEmail"
          name="replyEmail"
          type="email"
          component={component}
        />
        <Field
          label={<FormattedMessage id="contact.your-message" />}
          id="ProposalFormContactModal-message"
          name="message"
          type="textarea"
          rows={4}
          component={component}
        />
        <Field id="captcha" component={component} name="captcha" type="captcha" />
      </Modal.Body>
      <Modal.Footer>
        <CloseButton
          onClose={() => {
            onClose();
            reset();
          }}
        />
        <SubmitButton
          id="ProposalFormContactModal-submit"
          label="global.validate"
          isSubmitting={submitting}
          disabled={invalid || submitting || pristine}
          onSubmit={onSubmit}
        />
      </Modal.Footer>
    </form>
  </Modal>
);

const mapStateToProps = (state: State, { proposalId }: Props) => ({
  form: formName,
  initialValues: { proposalId },
  addCaptchaField: state.default.features.captcha,
});

const form = connect<any, any, _, _, _, _>(mapStateToProps)(
  reduxForm({
    validate,
    onSubmit,
    enableReinitialize: true,
  })(ProposalContactModal),
);

export default form;
