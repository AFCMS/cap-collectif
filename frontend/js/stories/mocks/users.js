// @flow

export const author = {
  id: 'user1',
  username: 'admin',
  media: {
    url: 'https://source.unsplash.com/random/150x150',
  },
  _links: {},
  vip: false,
  isViewer: false,
  isAdmin: true,
  isEvaluer: true,
};

export const followers = [
  {
    displayName: 'All Fred',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 4,
  },
  {
    displayName: 'guemas',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 0,
  },
  {
    displayName: 'Shezeboss',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 4,
  },
  {
    displayName: 'AS',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 240,
  },
  {
    displayName: 'Louisette',
    isEmailConfirmed: false,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 1,
  },
  {
    contributionsCount: 1,
    isEmailConfirmed: true,
    _links: null,
    media: null,
  },
  {
    displayName: 'guemas',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 0,
  },
  {
    displayName: 'Shezeboss',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 4,
  },
  {
    displayName: 'AS',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 240,
  },
];
