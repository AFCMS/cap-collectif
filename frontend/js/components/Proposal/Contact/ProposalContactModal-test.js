// @flow
/* eslint-env jest */
import * as React from 'react';
import { shallow } from 'enzyme';
import { ProposalContactModal } from './ProposalContactModal';
import { formMock } from '../../../mocks';

describe('<ProposalContactModal />', () => {
  it('renders correctly', () => {
    const props = {
      ...formMock,
      proposalId: 'proposalId324Z',
      authorName: 'Gertrude',
      onClose: jest.fn(),
      onSubmit: jest.fn(),
      addCaptchaField: false,
      show: false,
    };
    const wrapper = shallow(<ProposalContactModal {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
