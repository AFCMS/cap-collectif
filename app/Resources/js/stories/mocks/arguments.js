// @flow
import { author } from './users';

export const argument = {
  user: author,
  trashedStatus: '',
  createdAt: '1 mars 2016 à 00:00',
  publishedAt: '1 mars 2016 à 00:00',
  body:
    'In placeat reiciendis ut. Officiis praesentium quia minima ut tenetur officiis. Eaque fugit voluptates temporibus suscipit provident culpa culpa. Magni recusandae dolorem aut id.',
  votes: { totalCount: 50 },
};

export const argumentsData = [
  { ...argument, user: { ...argument.user, vip: true } },
  { ...argument },
  { ...argument, trashedStatus: 'INVISIBLE' },
  { ...argument },
];
