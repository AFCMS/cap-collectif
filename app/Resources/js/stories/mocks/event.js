// @flow
import { author } from './users';

export const event = {
  title: 'PHPTourDuTurfu',
  url: '#',
  image: {
    url: 'https://picsum.photos/300/400',
    alt: 'event picture',
  },
  date: {
    start: '2030-03-10 00:00:00',
    end: '2032-06-15 00:00:00',
  },
  location: {
    place: 'Tour Eiffel',
    zip: '75007',
    city: 'Paris',
  },
  category: 'Immobilier',

  user: author,
};
