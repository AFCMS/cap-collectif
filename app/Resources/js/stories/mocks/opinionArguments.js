// @flow
import { author } from './users';

export const opinionArgument = {
  user: author,
  trashedStatus: '',
  createdAt: '1 mars 2016 à 00:00',
  publishedAt: '1 mars 2016 à 00:00',
  body:
    'In placeat reiciendis ut. Officiis praesentium quia minima ut tenetur officiis. Eaque fugit voluptates temporibus suscipit provident culpa culpa. Magni recusandae dolorem aut id.',
  votes: { totalCount: 50 },
};

export const opinionArguments = [
  { ...opinionArgument, user: { ...opinionArgument.user, vip: true } },
  { ...opinionArgument },
  { ...opinionArgument, trashedStatus: 'INVISIBLE' },
  { ...opinionArgument },
];
