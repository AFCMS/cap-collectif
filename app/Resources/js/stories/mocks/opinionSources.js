// @flow
import { author } from './users';

export const opinionSource = {
  url: 'https://ui.cap-collectif.com/',
  user: author,
  createdAt: ' • 1 mars 2016',
  publishedAt: ' • 1 mars 2016',
  title: 'Source title',
  category: 'Politique',
  body:
    'In placeat reiciendis ut. Officiis praesentium quia minima ut tenetur officiis. Eaque fugit voluptates temporibus suscipit provident culpa culpa. Magni recusandae dolorem aut id.',
  votes: { totalCount: 2 },
};

export const opinionSources = [
  { ...opinionSource },
  { ...opinionSource },
  { ...opinionSource, user: { ...opinionSource.user, vip: true } },
  { ...opinionSource },
  { ...opinionSource, category: null },
];
