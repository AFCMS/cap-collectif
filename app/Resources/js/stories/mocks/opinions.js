// @flow
import { author } from './users';

export const opinion = {
  url: 'https://ui.cap-collectif.com/',
  title: 'Opinion',
  user: author,
  createdAt: ' • 1 mars 2018',
  pinned: true,
  publishedAt: ' • 1 mars 2018',
  votes: { totalCount: 4 },
  versions: { totalCount: 3 },
  arguments: { totalCount: 4 },
  sources: { totalCount: 0 },
  votesMitige: { totalCount: 1 },
  votesNok: { totalCount: 1 },
  votesOk: { totalCount: 2 },
};

export const opinions = [
  { ...opinion },
  { ...opinion, user: { ...opinion.user, vip: true } },
  { ...opinion, votes: { totalCount: 0 } },
];
