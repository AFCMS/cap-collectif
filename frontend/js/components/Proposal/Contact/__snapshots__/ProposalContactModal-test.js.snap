// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProposalContactModal /> renders correctly 1`] = `
<Modal
  animation={true}
  aria-labelledby="ProposalFormContactModal-modal"
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[MockFunction]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={false}
>
  <form
    id="ProposalContactModalForm"
    onSubmit={[MockFunction]}
  >
    <ModalHeader
      bsClass="modal-header"
      closeButton={true}
      closeLabel="Close"
    >
      <ModalTitle
        bsClass="modal-title"
        componentClass="h4"
        id="ProposalFormContactModal-title"
      >
        <FormattedMessage
          id="send-message-to"
          values={
            Object {
              "messageTo": "Gertrude",
            }
          }
        />
      </ModalTitle>
    </ModalHeader>
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    >
      <Field
        component={[Function]}
        id="ProposalFormContactModal-senderName"
        label={
          <FormattedMessage
            id="your-name"
            values={Object {}}
          />
        }
        name="senderName"
        type="text"
      />
      <Field
        component={[Function]}
        id="ProposalFormContactModal-message"
        label={
          <FormattedMessage
            id="contact.your-message"
            values={Object {}}
          />
        }
        name="message"
        rows={4}
        type="textarea"
      />
      <Field
        component={[Function]}
        id="ProposalFormContactModal-replyEmail"
        label={
          <FormattedMessage
            id="your-email-address"
            values={Object {}}
          />
        }
        name="replyEmail"
        type="email"
      />
      <Field
        component={[Function]}
        id="captcha"
        name="captcha"
        type="captcha"
      />
    </ModalBody>
    <ModalFooter
      bsClass="modal-footer"
      componentClass="div"
    >
      <CloseButton
        label="global.cancel"
        onClose={[MockFunction]}
      />
      <Connect(SubmitButton)
        disabled={false}
        id="ProposalFormContactModal-submit"
        isSubmitting={false}
        label="global.validate"
        onSubmit={[Function]}
      />
    </ModalFooter>
  </form>
</Modal>
`;
